flash/vision/video/classification/data.py

```diff
@@ -0,0 +1,170 @@
+
+
+class VideoPreprocessPreprocess(Preprocess):
+
+    def __init__(
+        self,
+        clip_sampler: ClipSampler,
+        train_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        val_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        test_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        predict_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+    ):
+
+        super().__init__()
+        self.clip_sampler = clip_sampler
+
+    def load_data(self, data: Any, dataset: IterableDataset) -> Dict:
+        return labeled_encoded_video_dataset(
+            data,
+            self.clip_sampler,
+        )
+
+
+class VideoClassificationData(DataModule):
+    """Data module for Video classification tasks."""
+
+    preprocess_cls = VideoPreprocessPreprocess
+
+    @classmethod
+    def instantiate_preprocess(
+        cls,
+        clip_sampler: ClipSampler,
+        video_sampler: Type[Sampler] = RandomSampler,
+        train_transform: Dict[str, Union[nn.Module, Callable]],
+        val_transform: Dict[str, Union[nn.Module, Callable]],
+        test_transform: Dict[str, Union[nn.Module, Callable]],
+        predict_transform: Dict[str, Union[nn.Module, Callable]],
+        preprocess_cls: Type[Preprocess] = None
+    ) -> Preprocess:
+        """
+        """
+        preprocess_cls = preprocess_cls or cls.preprocess_cls
+        preprocess: Preprocess = preprocess_cls(
+            clip_sampler, train_transform, val_transform, test_transform, predict_transform
+        )
+        return preprocess
+
+    @classmethod
+    def from_folders(
+        cls,
+        train_folder: Optional[Union[str, pathlib.Path]] = None,
+        val_folder: Optional[Union[str, pathlib.Path]] = None,
+        test_folder: Optional[Union[str, pathlib.Path]] = None,
+        predict_folder: Union[str, pathlib.Path] = None,
+        clip_sampler: Union[str, ClipSampler] = "random",
+        clip_duration: float = 2,
+        video_sampler: Type[Sampler] = RandomSampler,
+        clip_sampler_kwargs: Dict[str, Any] = None,
+        train_transform: Optional[Union[str, Dict]] = 'default',
+        val_transform: Optional[Union[str, Dict]] = 'default',
+        test_transform: Optional[Union[str, Dict]] = 'default',
+        predict_transform: Optional[Union[str, Dict]] = 'default',
+        batch_size: int = 4,
+        num_workers: Optional[int] = None,
+        preprocess_cls: Optional[Type[Preprocess]] = None,
+        **kwargs,
+    ) -> 'DataModule':
+        """
+
+        Creates a VideoClassificationData object from folders of images arranged in this way: ::
+
+            train/class_x/xxx.ext
+            train/class_x/xxy.ext
+            train/class_x/xxz.ext
+            train/class_y/123.ext
+            train/class_y/nsdf3.ext
+            train/class_y/asd932_.ext
```

Comment:
I'm not familiar with video dataset. Is this a common structure in the data ? /cc @tullie

Reply:
This follows TorchVision default structure.

Reply:
There's several formats of data we support:

1. loading encoded video paths from csv with format: "path_name, label" (PySlowFast repo uses this)
2. loading encoded video paths from directory structure as in comment (TorchVision uses this)
3. loading frame videos (i.e. a folder of images) from directory structure
4. loading frame videos from csv with a format specific to the dataset (e.g. EpicKitchen and DomSev datasets use this)